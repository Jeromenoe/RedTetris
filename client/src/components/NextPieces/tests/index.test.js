import React from 'react';
import { NextPieces } from '../index';
import renderer from 'react-test-renderer';
import configureStore from "redux-mock-store";
import {Provider} from 'react-redux';



describe('Testing NextPieces Component', () => {
	test('Next tetrominos', async () => {
		const mockStore = configureStore();
		const initialState = {
			me : {
				nextPieces: [
					[
						['I', 'I', 'I', 'I']
					],
					[
						['J', 0, 0, 0],
						['J', 'J', 'J', 0]
					]
				]
			}
		};
		const store = mockStore(initialState);
		const tree = renderer.create(<Provider store={store}><NextPieces /></Provider>).toJSON();
		expect(tree).toMatchSnapshot();
	  });
});