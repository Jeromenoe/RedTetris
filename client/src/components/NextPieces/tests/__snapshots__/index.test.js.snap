// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing NextPieces Component Next tetrominos 1`] = `
<div>
  <div
    className="sc-bdnxRM GaUJx"
  >
    <h1
      className="sc-hKFxyN kPweBh"
    >
      NEXT
    </h1>
    <div
      className="sc-gtsrHT kszIYe"
    >
      <div
        className="sc-dlnjwi kooEPI"
        height={1}
        width={4}
      >
        <div
          className="sc-eCApnc jJgApf"
          color="08a9a9"
          type="I"
        />
        <div
          className="sc-eCApnc jJgApf"
          color="08a9a9"
          type="I"
        />
        <div
          className="sc-eCApnc jJgApf"
          color="08a9a9"
          type="I"
        />
        <div
          className="sc-eCApnc jJgApf"
          color="08a9a9"
          type="I"
        />
      </div>
    </div>
    <div
      className="sc-gtsrHT kszIYe"
    >
      <div
        className="sc-dlnjwi jXvXiL"
        height={2}
        width={4}
      >
        <div
          className="sc-eCApnc kcZqRV"
          color="2471a3"
          type="J"
        />
        <div
          className="sc-eCApnc UOHvZ"
          color="e5e7e9"
          type="nextTetro"
        />
        <div
          className="sc-eCApnc UOHvZ"
          color="e5e7e9"
          type="nextTetro"
        />
        <div
          className="sc-eCApnc UOHvZ"
          color="e5e7e9"
          type="nextTetro"
        />
        <div
          className="sc-eCApnc kcZqRV"
          color="2471a3"
          type="J"
        />
        <div
          className="sc-eCApnc kcZqRV"
          color="2471a3"
          type="J"
        />
        <div
          className="sc-eCApnc kcZqRV"
          color="2471a3"
          type="J"
        />
        <div
          className="sc-eCApnc UOHvZ"
          color="e5e7e9"
          type="nextTetro"
        />
      </div>
    </div>
  </div>
</div>
`;
